### PR TITLE
Fix Fhir api response

### PR DESCRIPTION
**Pull request by sjpadgett**
- add scopes to sign in dialog
- some tuning in route parsing

### CODE REVIEW

**Comment by sjpadgett** [comment]
image

image




image

File: src/RestControllers/RestControllerHelper.php [hunk 1]
```diff
@@ -65,28 +65,50 @@ public static function validationHandler($validationResult)
...
+            $httpResponseBody = [];
+            if (!$processingResult->isValid()) {
+                http_response_code(400);
+                $httpResponseBody["validationErrors"] = $processingResult->getValidationMessages();
+            } elseif ($processingResult->hasInternalErrors()) {
+                http_response_code(500);
+                $httpResponseBody["internalErrors"] = $processingResult->getInternalErrors();
+            } else {
+                http_response_code($successStatusCode);
+                $dataResult = $processingResult->getData();
```
Comment (sjpadgett) [inline src/RestControllers/RestControllerHelper.php:85]: For now until I get Outcomes done which i'll need to refactor our Fhir apis to handle outcomes in resource. (need uris etc)

**Comment by sjpadgett** [comment]
@bradymiller 
Fhir success now gives:
```
{
  "id": "92328288-92ed-4b97-b15a-22ccc257a992",
  "meta": {
    "versionId": "1",
    "lastUpdated": "2020-12-20T00:31:55+00:00"
  },
  "resourceType": "Patient",
  "text": {}
...
}
```
Bundle is also okay.
And validation:
```
{
  "validationErrors": {
    "uuid": {
      "invalid or nonexisting value": " value 92328288-92ed-4b97-b15a-22ccc257a99"
    }
  }
}
```

Same for internals 
Looks like test only failing on fhir response changes.
Bringing in and if you don't get to tests i'll try some later. 
I've got other stuff need to get to.

**Comment by bradymiller** [comment]
nice!

tenor99

**Pull request merged**